packages/compiler/src/server/completion.ts

```diff
@@ -229,6 +262,33 @@ async function addRelativePathCompletion(
   }
 }
 
+function addModelCompletion(
```

Comment:
My idea here was to have a more generic solution for figuring out where we are in the document.
Today we have `getNodeAtPos()` which doesn't work super well when autocompleting. What I was wondering is if we could get more info in a single pass:
- parent node
- previous node
- next node
and if that would not be enough to always know where we are instead of having to dig back into the text

Reply:
Do you mean for the "{ ...}|" scenario so that we dont need to go back to document to check the cursor pos? then i think we would need some more details inside the node.

Reply:
yeah I am just wondering if we can get a more generic solution than having to check the raw text for this specific scenario

Reply:
added a PositionDetail interface to return exposing more information about the position besides the node